qa/L0_response_cache/ensemble_cache_test.py

```diff
@@ -0,0 +1,319 @@
+            )
+
+    def enable_response_cache_composing_model(self):
+        self._update_config(
+            self.composing_config_file,
+            self.response_cache_pattern,
+            self.response_cache_config,
+        )
+
+    def reset_config_files(self):
+        config_files = self._get_all_config_files()
+        for config_file in config_files:
+            self._remove_config(config_file, self.response_cache_config)
+            self._remove_config(config_file, self.decoupled_config)
+
+    def enable_decoupled_ensemble_model(self):
+        self._update_config(
+            self.ensemble_config_file,
+            self.decoupled_pattern,
+            self.decoupled_config,
+        )
+
+    def enable_decoupled_composing_model(self):
+        self._update_config(
+            self.composing_config_file, self.decoupled_pattern, self.decoupled_config
+        )
+
+
+class InferenceHandler:
+    def __init__(self):
+        self.config_manager = ModelConfigManager()
+        self.input_tensors = self._get_input_tensors()
+        self.triton_client = grpcclient.InferenceServerClient(
+            "localhost:8001", verbose=True
+        )
+
+    def _get_input_tensors(self):
+        input0_data = np.ones((1, 16), dtype=np.float32)
+        input1_data = np.ones((1, 16), dtype=np.float32)
+        input_tensors = [
+            grpcclient.InferInput(
+                "INPUT0", input0_data.shape, np_to_triton_dtype(input0_data.dtype)
+            ),
+            grpcclient.InferInput(
+                "INPUT1", input1_data.shape, np_to_triton_dtype(input0_data.dtype)
+            ),
+        ]
+        input_tensors[0].set_data_from_numpy(input0_data)
+        input_tensors[1].set_data_from_numpy(input1_data)
+        return input_tensors
+
+    def get_inference_statistics(self, model):
+        stats = self.triton_client.get_inference_statistics(
+            model_name=model, as_json=True
+        )
+        return stats["model_stats"][1]["inference_stats"]
+
+    def infer(self):
+        model = self.config_manager.get_ensemble_model()
+        output = self.triton_client.infer(model_name=model, inputs=self.input_tensors)
+        output0 = output.as_numpy("OUTPUT0")
+        output1 = output.as_numpy("OUTPUT1")
+        outputs = [output0, output1]
+        return outputs
+
+
+class InferenceValidator:
+    def __init__(self):
+        self.inference_handler = InferenceHandler()
+
+    def _check_valid_output(self, output):
+        assert output is not None, "Unexpected error in inference"
+
+    def _check_valid_output_inference(self, inference_outputs, cached_outputs):
+        assert np.array_equal(inference_outputs, cached_outputs), "Mismatched outputs"
+
+    def _check_zero_stats_baseline(self, model_inference_stats):
+        assert (
+            "count" not in model_inference_stats["success"]
+        ), "Unexpected error while retrieving statistics"
```

Comment:
I'd recommend picking a style and stick to it. Currently some asserts are followed by condition in brackets and some are not.

Reply:
I would also recommend adding expected and received values to the error msg, to ease debugging